src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -197,36 +251,26 @@ class _ConstraintsRepositoryPexRequest:
 
 
 @rule(level=LogLevel.DEBUG)
-async def pex_from_targets(request: PexFromTargetsRequest, python_setup: PythonSetup) -> PexRequest:
-    if request.direct_deps_only:
-        targets = await Get(Targets, Addresses(request.addresses))
-        direct_deps = await MultiGet(
-            Get(Targets, DependenciesRequest(tgt.get(Dependencies))) for tgt in targets
-        )
-        all_targets = FrozenOrderedSet(itertools.chain(*direct_deps, targets))
-    else:
-        transitive_targets = await Get(
-            TransitiveTargets, TransitiveTargetsRequest(request.addresses)
-        )
-        all_targets = transitive_targets.closure
+async def pex_from_targets(request: PexFromTargetsRequest) -> PexRequest:
+    interpreter_constraints = await Get(
+        InterpreterConstraints,
+        InterpreterConstraintsRequest(
+            request.addresses,
+            hardcoded_interpreter_constraints=request.hardcoded_interpreter_constraints,
```

Comment:
This is a key bit of this change: the logic for getting interpreter constraints and relevant targets was broken out of here into separate rules, and then used here.